Source/Controllers/SubviewControllers/SPTablesList.m

```diff
@@ -1937,6 +2058,83 @@ - (void) clearFilter
 	[listFilterField setStringValue:@""];
 }
 
+/**
+ * Reset the current pinned tables
+ */
+- (void)initPinnedTables {
+
+    [pinnedTables removeAllObjects];
+    NSString * hostName = [mySQLConnection host];
+    NSString * databaseName = [mySQLConnection database];
+    NSArray *tablesToPin = [_SQLitePinnedTableManager getPinnedTablesWithHostName:hostName databaseName:databaseName];
+    if (tablesToPin.count > 0) {
+        [tables insertObject:@"PINNED" atIndex:0];
+        [tableTypes insertObject:@(SPTableTypeNone) atIndex:0];
+        for (NSString *tableToPin in tablesToPin) {
+            [pinnedTables addObject:tableToPin];
+            SPTableType tableType = (SPTableType) [tableTypes[[tables indexOfObject:tableToPin]] integerValue];
+            [tables insertObject:tableToPin atIndex:1];
+            [tableTypes insertObject:@(tableType) atIndex:1];
+        }
+    }
+
+    [[tablesListView onMainThread] reloadData];
+    
+    // remove observer for previous hostname+database on database selection change by user
+    if (pinnedTableNotificationName != nil) {
+        [[NSNotificationCenter defaultCenter]
+            removeObserver:self
+            name:pinnedTableNotificationName
+            object:nil];
+    }
+    
+    // craft notification name such that notification goes only to tabs with same hostname+database
+    pinnedTableNotificationName = [NSString stringWithFormat:@"PinnedTableNotification-Host:%@-Database:%@", hostName, databaseName];
+    
+    [[NSNotificationCenter defaultCenter]
+        addObserver:self
+        selector:@selector(usePinTableNotification:)
+        name:pinnedTableNotificationName
+        object:nil];
+}
+
+
+- (void)usePinTableNotification:(nonnull NSNotification *) notification; {
```

Comment:
This is pretty cool and elegant actually, but I'd maybe rather think about trying to handle adds and removes by making the initPinnedTables function re-run safe and just reloading the pinned tables anytime there's a change in another db view? That might make the logic in initPinnedTables a little more complex, but would mean there aren't two chunks of code messing with the pinned tables section?